easybuild/easyblocks/c/cpmd.py

```diff
@@ -0,0 +1,208 @@
+
+        selected_base_config = None
+        selected_full_config = None
+        for cfc in config_file_candidates:
+            self.log.info("Trying configuration file: %s", cfc)
+            config_file_full = os.path.join(config_file_prefix, cfc)
+            if os.path.isfile(config_file_full):
+                selected_base_config = cfc
+                selected_full_config = config_file_full
+                self.log.info("Selected %s as base configuration file", cfc)
+                break
+
+        if selected_base_config is None:
+            raise EasyBuildError("Base configuration file does not exist. Please edit base_configuration or review the CPMD easyblock.")
+
+        try:
+            for line in fileinput.input(selected_full_config, inplace=1, backup='.orig'):
+                if self.toolchain.comp_family() in [toolchain.INTELCOMP]:
+                    ar_exe = "xiar -ruv"
+                    line = re.sub(r"^(\s*AR)=.*", r"\1='%s'" % ar_exe, line)
+                # Better to get CC and FC from the EasyBuild environment in this instance
+                line = re.sub(r"^(\s*CC=.*)",     r"#\1",        line)
+                line = re.sub(r"^(\s*FC=.*)",     r"#\1",        line)
+                #line = re.sub(r"^(\s*CFLAGS=)'?((?!\W-g\W)[^']*)'?\s*$", r"\1'{0}\2 '".format(os.getenv('CFLAGS')), line)
+                line = re.sub(r"^(\s*LD)=.*",     r"\1='$(FC)'", line)
+                sys.stdout.write(line)
+        except IOError, err:
+            raise EasyBuildError("Failed to patch %s: %s", selected_base_config, err)
+
+        if self.cfg['configure_cmd_prefix']:
+            if cmd_prefix:
+                tup = (cmd_prefix, self.cfg['configure_cmd_prefix'])
+                self.log.debug("Specified cmd_prefix '%s' is overruled by configure_cmd_prefix '%s'" % tup)
+            cmd_prefix = self.cfg['configure_cmd_prefix']
+
+        if self.cfg['tar_config_opts']:
+            # setting am_cv_prog_tar_ustar avoids that configure tries to figure out
+            # which command should be used for tarring/untarring
+            # am__tar and am__untar should be set to something decent (tar should work)
+            tar_vars = {
+                'am__tar': 'tar chf - "$$tardir"',
+                'am__untar': 'tar xf -',
+                'am_cv_prog_tar_ustar': 'easybuild_avoid_ustar_testing'
+            }
+            for (key, val) in tar_vars.items():
+                self.cfg.update('preconfigopts', "%s='%s'" % (key, val))
+
+        options = [self.cfg['configopts']]
+
+        # enable OpenMP support if desired
+        if self.toolchain.options.get('openmp', None):
+            options.append("-omp")
+
+        # This "option" has to come last as it's the chief argument, coming after
+        # all flags and so forth.
+        options.append(selected_base_config)
+
+        cmd = "%(preconfigopts)s %(cmd_prefix)s./configure.sh %(prefix_opt)s%(installdir)s %(configopts)s" % {
```

Comment:
Because the superclass configure_step method also does a bunch of other things I don't want.